app/questionnaire/navigation.py

```diff
@@ -141,7 +141,7 @@ def _generate_repeated_items(self, link_names, first_location, completed_id, no_
             location.group_instance = i
             is_current_group_instance = is_current_group and i == current_group_instance
 
-            if link_names:
+            if link_names and link_names.get(i):
```

Comment:
Yeah, that's being being fixed in (941). In theory, you should never reach a situation where this fix is required. But actually in the case where the Navigation can't construct a name for the link, something is probably wrong, so this change just ensures that an invalid link is displayed in the navigation.